p2p/dht/bootstrap.go

```diff
@@ -49,6 +51,16 @@ func (d *KadDHT) Bootstrap() error {
 		return ErrConnectToBootNode
 	}
 
+	ExitApp := make(chan bool, 1)
+	signalChan := make(chan os.Signal)
+	signal.Notify(signalChan, os.Interrupt)
```

Comment:
We already register to the Interrupt event in SpacemeshApp. Instead of registering again you should inject a [context with cancel](https://golang.org/pkg/context/#example_WithCancel) and let it notify KadDHT whenever it should halt

Reply:
@zalmen basically will need to inject Context from the `func main()` to here. Is that correct? according to best practice should modify all functions to take context as first parameter `func DoSomething(ctx context.Context, arg Arg)`.

Reply:
I updated the code to use the Context with cancel. 
There are two test that are failing `github.com/spacemeshos/go-spacemesh/p2p/net` and 
`github.com/spacemeshos/go-spacemesh/app` I am looking into it now.